src/databases/Lines/avtLinesFileFormat.C

```diff
@@ -67,6 +63,31 @@ avtLinesFileFormat::~avtLinesFileFormat()
     }
 }
 
+// ****************************************************************************
+//  Method: avtLinesFileFormat::FreeUpResources
+//
+//  Purpose:
+//      When VisIt is done focusing on a particular timestep, it asks that
+//      timestep to free up any resources (memory, file descriptors) that
+//      it has associated with it.  This method is the mechanism for doing
+//      that.
+//
+//  Programmer: Philip Blakely
+//  Creation:   Mon Jun 22 15:29:05 PDT 2020
+//
+// ****************************************************************************
+
+void
+avtLinesFileFormat::FreeUpResources(void)
+{
+    for (size_t i = 0 ; i < lines.size() ; i++)
+    {
+        lines[i]->Delete();
+    }
+    
```

Comment:
Should clear() also be called on this vector after deleting memory to make sure that there aren't any attempts to access it later, or are there already mechanisms preventing this?

Reply:
Good point @aowen87. I think a call to clear() is needed.

Reply:
I think the clear in the destructor is unnecessary, since presumably nothing will be called after the destructor and the destructure will call the vector destructor. I do think having a clear in FreeUpResources would be good though, since now you have a vector with pointers to invalid memory.

Reply:
@brugger1 ...yeah, I meant to add it in FreeUpResources and accidentally put it in the destructor. Need to stop multitasking!